tensorflow/compiler/xrt/xrt_memory_manager.h

```diff
@@ -116,7 +118,8 @@ class XRTMemoryManager : public ResourceBase {
   template <typename T>
   xla::StatusOr<T> Run(const std::function<xla::StatusOr<T>()>& runfn,
                        xla::Backend* backend, int device_ordinal,
-                       size_t requested_free_size);
+                       size_t requested_free_size,
+                       se::DeviceMemoryAllocator* allocator = nullptr);
```

Comment:
Instead of only defaulting `allocator` to nullptr for functions that is used by TPU ops, maybe we should do that for all xrt functions(that you added `allocator` parameter to). We don't need to nullptr check `allocator` for all functions through, we can only do that for the lowest level function like `AllocateScopedShapedBuffer`.

This way we make it clear that allocator is not required for any of these functions. What do you think?

Reply:
Yeah, then the nullptr checks could converge to `XRTTupleAllocation::CreateFromBuffer` and `XRTTupleAllocation::SwapIn`